scripts/artifacts-building/README.md

```diff
@@ -0,0 +1,115 @@
+were built.
+
+1. Apt artifacts are built. This is done per RPC-O series and published
+   using the value for ``rpc_release`` and the distribution code name.
+   Current apt repositories available can be found on
+   [rpc-repo](http://rpc-repo.rackspace.com/apt-mirror/integrated/dists/).
+
+   The upstream apt repositories to mirror are all set in
+   ``scripts/artifacts-building/apt/aptly-vars.yml`` in the ``aptly_mirrors``
+   variable. They are combined into snapshots per release, then combined into
+   an integrated snapshot as specified in the variable ``aptly_miko_mapping``.
+
+   Apt repositories which provide the same package name across two different
+   distributions with sdifferent contents have to be published as independent
+   repositories. Those are defined in the ``aptly_n_mapping`` variable.
+
+   Only one Apt artifact build can be executed at a time across all
+   series/tests as the test executes on a long running host which has
+   aptly installed. Aptly's CLI locks the database and can therefore only
+   execute one job at a time. The job executes
+   ``scripts/artifacts-building/apt/build-apt-artifacts.sh``.
+
+2. Git artifacts are built. This is per RPC-O series and published using
+   the value for ``rpc_release``. Published artifacts can be found on
+   [rpc-repo](http://rpc-repo.rackspace.com/git-archives/). The job
+   executes ``scripts/artifacts-building/git/build-git-artifacts.sh``.
+
+3. Python artifacts are built using the previously built apt artifacts.
+   This is necessary as some wheels are linked to specific C libraries
+   and therefore the matching of a specific wheel version to a specific
+   apt package is important. This is done using a partial RPC-O AIO (only
+   the repo server is built) using the standard repo-build playbook from
+   OSA, but with selective git cloning, selective wheel building and
+   selective venv building disabled so that we have packages and venvs
+   for everything supported by RPC-O. Published artifacts can be found
+   on [rpc-repo](http://rpc-repo.rackspace.com/os-releases/). The job
+   executes ``scripts/artifacts-building/python/build-python-artifacts.sh``.
+
+4. Container artifacts are built using the previously built apt and python
+   artifacts. This is done by preparing the host as a LXC host, then building
+   a series on containers in turn without starting them but instead executing
+   the appropriate roles against them using a chroot connection and tagfilter
+   strategy. The tagfilter strategy ignores all ``-config`` tasks and filters
+   out all notifiers so that the chroot only has the right apt packages
+   installed and the right python packages and venvs installed. Published
+   artifacts can be found on
+   [rpc-repo](http://rpc-repo.rackspace.com/lxc-images/). The job executes
+   ``scripts/artifacts-building/containers/build-process.sh``.
+
+As RPC-O executes deploy tests in PR's where the value for ``rpc_release`` is
```

Comment:
That is unclear.

"For PRs, tests are also implemented in parallel so that they give results back quickly, whereas the periodic jobs are set to run in the proper sequence. We implemented a series of check to verify if previous artifacts exist for the given release.

For PRs, the test process will reuse the required artifacts for each build if possible.
Example, container artifact building test will ensure we can still build container artifacts, based on existing python/apt artifacts. Python artifacts will be built based on existing apt artifacts. If no artifact exist for this ``rpc_release``, new ones will be built. "

We can keep the:
"
The PR jobs are all read-only (they do not upload to rpc-repo), but the
periodic jobs will upload to rpc-repo if artifacts for the given release do
not exist yet. To replace artifacts for a series, execute the periodic job
for the series with ``REPLACE_ARTIFACTS`` set to ``YES``.
"

Reply:
I've clarified this in a different way.